### PR TITLE
Fr 64 create api methods for donations

### PR DESCRIPTION
**Pull request by VladimirKarin**
https://vladimirkarin.atlassian.net/browse/FR-64

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,6 +35,29 @@ app.get('/ideas', (req, res) => {
...
+});
+
+app.get('/donations/sumDifference', (req, res) => {
+    res.status(200).json(ideasSumDifference(req.body.ideaId));
+});
+
+app.post('/donations', (req, res) => {
+    res.status(200)
+        .json(
+            createDonationsWithDataValidation(
```
Comment (fedosejev) [inline index.js:51]: Can `createDonationsWithDataValidation` function throw an error?

Comment (VladimirKarin) [reply index.js]: Updated and tested.

**Comment by fedosejev** [comment]
Please resolve conflicts

**Comment by VladimirKarin** [comment]
Conflicts resolved.

File: index.js [hunk 2]
```diff
@@ -107,6 +112,30 @@ app.delete('/ideas', (req, res) => {
...
+        createDonationsWithDataValidation(
+            req.body.firstName,
+            req.body.sum,
+            req.body.userId,
+            req.body.ideaId
+        );
+    } catch (Error) {
+        res.status(404).send(Error.message);
+    }
+    res.status(200).send('Donation created successfully');
```
Comment (fedosejev) [inline index.js:136]: I've noticed you're using 3 message formats:
1. Without `.`
2. With `.`
3. With `..`

Can you format all messages consistently please?

Comment (VladimirKarin) [reply index.js]: Yes, I've formated. You are totally right.
And I've added some error messages in business logis\donations.js functions (where IdeaId was mentioned)

File: index.js [hunk 3]
```diff
@@ -107,6 +112,30 @@ app.delete('/ideas', (req, res) => {
     res.status(200).send('Ideas successfully deleted.');
 });
 
+//DONATION METHODS
+
+app.get('/donations', (req, res) => {
+    res.status(200).json(ideasDonationSum(req.body.ideaId));
+});
+
+app.get('/donations/sumDifference', (req, res) => {
```
Comment (fedosejev) [inline index.js:121]: What does this API return?

Comment (VladimirKarin) [reply index.js:121]: It return difference of askedSum and totalDonationSum (for this idea)

File: index.js [hunk 4]
```diff
@@ -107,6 +112,30 @@ app.delete('/ideas', (req, res) => {
...
+    res.status(200).json(ideasSumDifference(req.body.ideaId));
+});
+
+app.post('/donations', (req, res) => {
+    try {
+        createDonationsWithDataValidation(
```
Comment (fedosejev) [inline index.js:127]: Can this function be named `createDonation`?

File: index.js [hunk 5]
```diff
@@ -107,6 +112,30 @@ app.delete('/ideas', (req, res) => {
...
+            req.body.firstName,
+            req.body.sum,
+            req.body.userId,
+            req.body.ideaId
+        );
+    } catch (Error) {
```
Comment (fedosejev) [inline index.js:133]: Should be `error`.

`Error` is a global JavaScript function.

File: index.js (hunk 4 above)
Comment (VladimirKarin) [reply index.js]: As long as I have hands it can! Updated. Commited.

File: index.js (hunk 5 above)
Comment (VladimirKarin) [reply index.js]: updated and commited

File: businessRules/donations.js [hunk 6]
```diff
@@ -1,8 +1,16 @@
...
-const { updateIdea, getAllIdeas } = require('../models/ideas');
+const { getAllIdeas } = require('../models/ideas');
+const { getIdeas } = require('../utils/storage');
+
+function createDonation(firstName, sum, userId, ideaId) {
+    let ideas = getIdeas();
+    const idea = ideas.find((idea) => ideaId === idea.id);
+
+    if (!idea) {
+        throw new Error(`Error. No idea with such ID found.`);
```
Comment (fedosejev) [inline businessRules/donations.js:11]: No need to use ` here - you can use '

Comment (fedosejev) [reply businessRules/donations.js:11]: Same for all other error messages.

Comment (VladimirKarin) [reply businessRules/donations.js]: fixed that one and commited.

**Review by fedosejev — APPROVED** [review]

**Pull request merged**